### PR TITLE
[NEAT-106] Skip dumping default space/prefix/version. 

### PR DESCRIPTION
**Pull request by doctrino**
Implemented custom serialization for all rules. Why

* Ability to remove default prefix/space/version
* Automatically add role to metadata.
* Hide the implementation detail that we use UserList, e.g. `properties: {"data": [...]}` and instead just `properties: [...]`.

No more default space/prefix/version in the YAML or Excel Exporters:
image

### CODE REVIEW

File: cognite/neat/rules/exporters/_rules2yaml.py [hunk 1]
```diff
@@ -69,11 +69,10 @@ def export(self, rules: Rules) -> str:
         rules = self._convert_to_output_role(rules, self.output_role)
         # model_dump_json ensures that the output is in JSON format,
         # if we don't do this, we will get Enums and other types that are not serializable to YAML
-        json_output = json.loads(rules.model_dump_json())
-        json_output["metadata"]["role"] = rules.metadata.role.value
```
Comment (doctrino) [inline cognite/neat/rules/exporters/_rules2yaml.py:73]: No longer necessary, as it is done automatically by the serializer

File: cognite/neat/rules/exporters/_rules2excel.py [hunk 2]
```diff
@@ -63,29 +64,37 @@ def export(self, rules: Rules) -> Workbook:
         # Remove default sheet named "Sheet"
         workbook.remove(workbook["Sheet"])
 
+        dumped_rules = rules.model_dump(by_alias=True)
+
         metadata_sheet = workbook.create_sheet("Metadata")
-        metadata_sheet.append(["role", rules.metadata.role.value])
```
Comment (doctrino) [inline cognite/neat/rules/exporters/_rules2excel.py:67]: No longer necessary, automatically handled.

File: cognite/neat/rules/models/_rules/information_rules.py [hunk 3]
```diff
@@ -268,6 +269,41 @@ def validate_class_has_properties_or_parent(self) -> Self:
...
+            dumped = prop.model_dump(**kwargs)
+            for field_name in field_names:
+                if value := dumped.get(field_name):
+                    dumped[field_name] = value.removeprefix(default_prefix)
+            properties.append(dumped)
+
+        field_names = ["Class"] if info.by_alias else ["class_"]
+        classes = []
+        parent_name = "Parent Class" if info.by_alias else "parent"
+        for prop in self.properties:
```
Comment (nikokaoja) [inline cognite/neat/rules/models/_rules/information_rules.py:295]: I think this is error, you should iterate over `self.classes` not `self.properties`

**Review by nikokaoja — COMMENTED** [review]
Couple of things to fix, main comments:
- a bit long serializer for DMS architect
- adding a method to return filed names by alias or not would be useful

**Review by nikokaoja — APPROVED** [review]

File: cognite/neat/rules/models/_rules/information_rules.py (hunk 3 above)
Comment (doctrino) [reply cognite/neat/rules/models/_rules/information_rules.py:295]: You are right

File: cognite/neat/rules/models/_rules/base.py [hunk 4]
```diff
@@ -171,6 +171,10 @@ def mandatory_fields(cls, use_alias=False) -> set[str]:
         """Returns a set of mandatory fields for the model."""
         return _get_required_fields(cls, use_alias)
 
+    @model_serializer(mode="wrap")
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite/neat/rules/models/_rules/base.py:174]: ??? It is a serializer it is called everytime someones dump metadata

**Pull request merged**